s3-proxy/nginx.conf

```diff
@@ -112,6 +112,60 @@ http {
             proxy_hide_header X-Archive-Files;
         }
 
+        location ~ ^/browse/.* {
+            proxy_pass $registry;
+            proxy_http_version 1.1;
+            proxy_buffering off;
+            proxy_ssl_verify on;
+            proxy_ssl_verify_depth 5;
+            proxy_ssl_trusted_certificate /etc/nginx/certs.pem;
+            proxy_ssl_name $REGISTRY_HOST;
+
+            proxy_set_header Host $REGISTRY_HOST;
+
+            # proxy_set_header Connection "";
+            # proxy_cache_bypass $http_pragma;
+            # proxy_redirect off;
+
+            # proxy_set_header X-Real-URL $upstream_http_location;
+            # resolver 8.8.8.8;
+
+            # add_header 'Access-Control-Allow-Origin' '*';
+            #
+            # proxy_set_header 'Content-Type' 'text/html';
+            # add_header 'Content-Type' 'text/html';
+
+            proxy_intercept_errors on;
+            error_page 301 302 307 = @handle_redirect;
+
+            # if ($upstream_http_location) {
+            #     proxy_pass $upstream_http_location;
+            # }
+        }
+
+        location @handle_redirect {
+            internal;
+            # TODO: this must be forced only for html files or, better yet,
+            # not forced at all -- we need to provide proper content-type headers
+            # when signing the urls in the registry
+            add_header 'Content-Type' 'text/html';
```

Comment:
I'd expect .html files to already have the correct header - e.g., if you use `aws s3 cp foo.html s3://whatever/`, it'll set the content type. But if not, then yeah, doing it through signed URLs should work,